File: astro/extractedcode/example-client-libraries/python/add-magic-link.py
from fusionauth.fusionauth_client import FusionAuthClient
import os
import sys

APPLICATION_ID = "e9fdb985-9173-4e01-9d73-ac2d60d1dc8e";

#  You must supply your API key 
api_key_name = 'fusionauth_api_key'
api_key = os.getenv(api_key_name)
if not api_key:
  sys.exit("please set api key in the '" + api_key_name + "' environment variable")

client = FusionAuthClient(api_key, 'http://localhost:9011')

client_response = client.patch_application(APPLICATION_ID, {"application": {"passwordlessConfiguration":{"enabled":True}}})
if not client_response.was_successful():
  sys.exit("couldn't update application "+ str(client_response.error_response))

File: astro/extractedcode/example-client-libraries/java/src/main/java/io/fusionauth/example/AddMagicLink.java
package io.fusionauth.example;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

import com.inversoft.error.Errors;
import com.inversoft.rest.ClientResponse;

import io.fusionauth.client.FusionAuthClient;
import io.fusionauth.domain.api.ApplicationResponse;

public class AddMagicLink {

  public static void main(String[] args) {
    final String apiKey = System.getProperty("fusionauth.api.key");
    final FusionAuthClient client = new FusionAuthClient(apiKey, "http://localhost:9011");

    // enable magic links
    UUID clientId = UUID.fromString(Setup.APPLICATION_ID);

    Map<String, Object> applicationMap = new HashMap<String, Object>();
    Map<String, Object> passwordlessConfigurationMap = new HashMap<String, Object>();
    Map<String, Object> enableMagicLinkUpdateMap = new HashMap<String, Object>();

    passwordlessConfigurationMap.put("enabled",true);
    applicationMap.put("passwordlessConfiguration", passwordlessConfigurationMap);
    enableMagicLinkUpdateMap.put("application", applicationMap);

    ClientResponse<ApplicationResponse, Errors> applicationResponse = client.patchApplication(clientId, enableMagicLinkUpdateMap);
    if (!applicationResponse.wasSuccessful()) {
      throw new RuntimeException("couldn't update application");
    }
  }
}
